test/logger.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Logger, logger} from '@google-cloud/common';
+
+const PASS_THROUGH_LOG_LEVEL = Number(process.env.GCLOUD_TEST_LOG_LEVEL || 0);
+
+// tslint:disable-next-line:no-any
+type LoggerFunction = (message: any, ...args: any[]) => void;
+
+export class TestLogger implements Logger {
+  private logs:
+      {[k in keyof Logger]:
+           string[]} = {error: [], warn: [], info: [], debug: [], silly: []};
+  private innerLogger = logger({level: logger.LEVELS[PASS_THROUGH_LOG_LEVEL]});
+
+  private makeLoggerFn(logLevel: keyof Logger): LoggerFunction {
+    // TODO(kjin): When we drop support for Node 4, use spread args.
```

Comment:
Doesn't TS compiler convert to using `arguments`? That's what I'm observing in other projects.

Reply:
Possibly only when using `target: 'es5'`. We currently need tests to be compiled to ES6 for generators (granted the use of generators is probably constrained to a single test).

Reply:
Yes, only when the target is es5.

How does the test that uses generators run for Node 4 then?

It'd be great if the TS compiler provided a way to specify a different target for each Node version, but I don't think it supports that now.

Reply:
Node 4 supports generators, so it's alright to compile to ES6 as long as we don't use things like destructuring or spread args.

I agree, the only alternative is having separate config files which isn't very clean.